frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditActions/ActionsEditor/ActionsEditor.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioCard,
+  StudioTabs,
+  StudioDivider,
+  StudioButton,
+  StudioDeleteButton,
+  StudioProperty,
+  StudioParagraph,
+} from '@studio/components';
+import { CheckmarkIcon } from '@studio/icons';
+import { CustomActions } from './CustomActions';
+import { PredefinedActions } from './PredefinedActions';
+import { useBpmnContext } from '../../../../../contexts/BpmnContext';
+import { type Action, BpmnActionModeler } from '../../../../../utils/bpmn/BpmnActionModeler';
+import { getPredefinedActions, isActionRequiredForTask } from '../../../../../utils/processActions';
+
+import classes from './ActionsEditor.module.css';
+
+enum TabIds {
+  Predefined = 'predefined',
+  Custom = 'custom',
+}
+
+type ComponentMode = 'edit' | 'view';
+
+export type ActionsEditorProps = {
+  actionElement: Action;
+  actionIndex: number;
+  mode?: ComponentMode;
+};
+export const ActionsEditor = ({
+  actionElement,
+  actionIndex,
+  mode,
+}: ActionsEditorProps): React.ReactElement => {
+  const [componentMode, setComponentMode] = React.useState<ComponentMode>(mode || 'view');
+  const { t } = useTranslation();
+  const { bpmnDetails } = useBpmnContext();
+  const bpmnActionModeler = new BpmnActionModeler(bpmnDetails.element);
+
+  const actionLabel = t('process_editor.configuration_panel_actions_action_label', {
+    actionIndex: actionIndex + 1,
+    actionName: actionElement.action,
+  });
+
+  if (componentMode === 'edit') {
+    return (
+      <ActionEditable
+        actionElement={actionElement}
+        actionIndex={actionIndex}
+        onClose={() => setComponentMode('view')}
+        onDelete={() => bpmnActionModeler.deleteActionFromTask(actionElement)}
+      />
+    );
+  }
+
+  return (
+    <StudioProperty.Button
+      aria-label={actionLabel}
+      readOnly={isActionRequiredForTask(actionElement.action, bpmnDetails.taskType)}
+      onClick={() => setComponentMode('edit')}
+      property={actionLabel}
+      value={actionElement.action}
+      className={classes.actionView}
+    />
+  );
+};
+
+type ActionEditableProps = {
+  actionElement: Action;
+  actionIndex: number;
+  onClose: () => void;
+  onDelete: () => void;
+};
+const ActionEditable = ({
```

Comment:
Should we move this to a separate file?

Reply:
I choosen to adding it to the same file because `ActionEditable` is tightly coupled with `ActionsEditor`. 😊